PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+    "*.js": ["@parcel/optimizer-uglify"],
+    "*.css": ["@parcel/optimizer-cssnano"],
+    "*.html": ["@parcel/optimizer-htmlnano"],
+    "*.{png,jpg,jpeg,svg,...}": ["@parcel/optimizer-imagemin"]
+  },
+  "loaders": {
+    "*.js": "@parcel/loader-js",
+    "*.wasm": "@parcel/loader-wasm"
+  },
+  "reporters": ["@parcel/reporter-detailed"]
+}
+```
+
+#### Glob maps in `.parcelrc`
+
+Many config properties like `transforms` or `packagers` use objects as maps of
+globs to package names. While objects in JSON are technically unordered, Parcel
+does use the order to give globs priority when a file name is being tested
+against them.
+
+```json
+{
+  "transforms": {
+    "icons/*.svg": ["highest-priority"],
+    "*.svg": ["lowest-priority"]
+  }
+}
+```
+
+Here if we are trying to find a transform for the file `icons/home.svg`, we'll
+work our way down the globs until we find a match, which would be
+`icons/*.svg`, we never reach `*.svg`.
+
+#### `.parcelrc#extends`
+
+`extends` can either be a string or an array of strings that specify base
+configs to extend. That base configuration can be the path to another
+`.parcelrc` file or the name of a Parcel config package.
+
+```json
+{
+  "extends": "@parcel/config-default",
+  "extends": "../.parcelrc",
+  "extends": ["@parcel/config-default", "@company/parcel-config"]
+}
+```
+
+When extending a config, Parcel does a shallow merge of the two configs.
+
+#### `.parcelrc#resolvers`
+
+`resolvers` is an array of strings that specifies the name of a Parcel resolver
+package.
+
+```json
+{
+  "resolvers": ["@parcel/resolver-default"]
+}
+```
+
+See [Resolvers](#resolvers)
+
+#### `.parcelrc#transforms`
+
+`transforms` is an object map of globs to arrays of Parcel transform packages.
+
+```json
+{
+  "transforms": {
+    "*.js": ["@parcel/transform-babel"]
+  }
+}
+```
+
+See [Transforms](#transforms)
+
+#### `.parcelrc#bundler`
+
+`bundler` is a string that specifies the name of a Parcel bundler package.
+
+```json
+{
+  "bundler": "@parcel/bundler-v1"
+}
+```
+
+See [Bundlers](#bundlers)
+
+#### `.parcelrc#packagers`
+
+`packagers` is an object map of globs to Parcel packager packages.
+
+```json
+{
+  "packagers": {
+    "*.js": ["@parcel/packager-js"]
+  }
+}
+```
+
+See [Packagers](#packagers)
+
+#### `.parcelrc#optimizers`
+
+`optimizers` is an object map of globs to arrays of Parcel optimizer packages.
+
+```json
+{
+  "optimizers": {
+    "*.js": ["@parcel/optimizer-uglify"]
+  }
+}
+```
+
+See [Optimizers](#optimizers)
+
+#### `.parcelrc#loaders`
+
+`loaders`  is an object map of globs to Parcel loader packages. See
+[Loaders](#).
+
+```json
+{
+  "loaders": {
+    "*.js": "@parcel/loader-js",
+    "*.wasm": "@parcel/loader-wasm"
+  }
+}
+```
+
+See [Loaders](#loaders)
+
+#### `.parcelrc#reporters`
+
+`reporters` is an array of Parcel reporter packages. See [Reporters](#).
+
+```json
+{
+  "reporters": ["@parcel/reporter-detailed"]
+}
+```
+
+See [Reporters](#reporters)
+
+## Parcel Architecture
```

Comment:
An in-memory FS was discussed before. Will Parcel 2 maintain an in-memory FS? Will plugins get access to that FS?

I'm asking because some tools like GraphQL codegen write `.d.ts` TypeScript files to disk to make TypeScript happy about types. if the TypeScript compiler uses an in-memory FS and GraphQL writes to that you won't need to write to **source** directories to make the build pass